File: trees/783. Minimum Distance Between BST Nodes/minDiffInBST.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def minDiffInBST(self, root):
        """
        Finds the minimum distance in the BST using inorder traversal (since it gives the values in sorted order),
        by keeping track of the minimum distance while traversing the BST

        :type root: Optional[TreeNode]
        :rtype: int
        """
        current = root
        stack = []
        minimum = float('inf')
        prev = None

        while current or stack:
            while current:
                stack.append(current)
                current = current.left
            
            current = stack.pop()
            if prev:
                if current.val - prev.val < minimum:
                    minimum = current.val - prev.val
            
            prev = current
            current = current.right
        
        return minimum

File: trees/783. Minimum Distance Between BST Nodes/minDiffInBST.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDiffInBST = function(root) {
    let current = root;
    let stack = [];
    let min = Infinity;
    let prev = null;

    while (current || stack.length > 0) {
        while (current) {
            stack.push(current);
            current = current.left;
        }

        current = stack.pop();

        if (prev) {
            if (current.val - prev.val < min) {
                min = current.val - prev.val;
            }
        }

        prev = current;
        current = current.right;
    }

    return min;
};